wetterdienst/dwd/radar/access.py

```diff
@@ -0,0 +1,447 @@
+                    ]
+
+                # This is for matching historical data, e.g. "RW-200509.tar.gz".
+                else:
+                    file_index = file_index[
+                        (
+                            file_index[DWDMetaColumns.DATETIME.value].dt.year
+                            == start_date.year
+                        )
+                        & (
+                            file_index[DWDMetaColumns.DATETIME.value].dt.month
+                            == start_date.month
+                        )
+                    ]
+
+                results.append(file_index)
+
+            file_index = pd.concat(results)
+
+            if file_index.empty:
+                log.warning(f"No radar file found for {parameter}, {site}, {format}")
+                return
+
+            # Iterate list of files and yield "RadarResult" items.
+            for _, row in file_index.iterrows():
+                url = row[DWDMetaColumns.FILENAME.value]
+                yield download_radolan_data(start_date, url)
+
+        else:
+            file_index = create_fileindex_radar(
+                parameter=parameter,
+                site=site,
+                format=format,
+                subset=subset,
+                parse_datetime=True,
+            )
+
+            # Filter for dates range if start_date and end_date are defined.
+            file_index = file_index[
+                (file_index[DWDMetaColumns.DATETIME.value] >= start_date)
+                & (file_index[DWDMetaColumns.DATETIME.value] < end_date)
+            ]
+
+            if file_index.empty:
+                log.warning(f"No radar file found for {parameter}, {site}, {format}")
+                return
+
+            # Iterate list of files and yield "RadarResult" items.
+            for _, row in file_index.iterrows():
+                date_time = row[DWDMetaColumns.DATETIME.value]
+                url = row[DWDMetaColumns.FILENAME.value]
+
+                for result in _download_generic_data(url=url):
+                    if result.timestamp is None:
+                        result.timestamp = date_time
+                    yield result
+
+
+def should_cache_download(*args, **kwargs) -> bool:  # pragma: no cover
+    """
+    Determine whether this specific result should be cached.
+
+    Here, we don't want to cache any files containing "-latest-" in their filenames.
+
+    :param args: Arguments of decorated function.
+    :param kwargs: Keyword arguments of decorated function.
+    :return: When cache should be dimissed, return False. Otherwise, return True.
+    """
+    url = args[0]
+    if "-latest-" in url:
+        return False
+    return True
+
+
+@payload_cache_five_minutes.cache_on_arguments(should_cache_fn=should_cache_download)
+def _download_generic_data_cached(url: str) -> BytesIO:
+    return url, download_file_from_dwd(url)
+
+
+def _download_generic_data(url: str) -> Generator[RadarResult, None, None]:
```

Comment:
Why is url a `str` and not a `Path` object as we use in different other functions? 
Path has built in functions to check out suffix e.g.

Reply:
Using Path objects for operating on URLs hasn't always been without hassle in the past. So, we gradually shifted towards strings and URL operations using `urljoin()` instead of using Path objects at all. See also

https://github.com/earthobservations/wetterdienst/blob/643737573a1b867540b4079ebacd480bf850b933/wetterdienst/util/network.py#L9-L49